watcher/lib/influxdb/rollups.go

```diff
@@ -78,12 +122,12 @@ type Function struct {
 
 // Check verifies the function configuration is correct
 func (f Function) Check() error {
-	if !utils.OneOf(f.Function, constants.SimpleFunctions) && !isCompositeFunc(f) {
+	if !utils.OneOf(f.Function, constants.SimpleFunctions) && !f.isCompositeFunc() {
 		return trace.BadParameter(
 			"invalid Function, must be one of %v, or a composite function starting with one of %v prefixes",
 			constants.SimpleFunctions, constants.CompositeFunctions)
 	}
-	if isCompositeFunc(f) {
+	if f.isCompositeFunc() {
```

Comment:
Here, the code completely deviates at the point where you check if the function is composite - invert the condition to keep it left-aligned:
```go
func (f *Function) buildFunction() (string, error) {
	alias := f.Alias
	if alias == "" {
		alias = f.Field
	}

	// split function name, based on the "_" separator (eg: percentile_99, top_10, ecc)
	err := f.Check()
	if err != nil {
		return "", trace.Wrap(err)
	}

	if !f.isComposite() {
		return fmt.Sprintf(`%v("%v") as %v`, f.Function, f.Field, alias), nil
	}

	funcAndValue := strings.Split(f.Function, "_")
	funcName := funcAndValue[0]
	param := funcAndValue[1]

	err := validateParam(funcName, param)
	if err != nil {
		return "", trace.Wrap(err)
	}
	return fmt.Sprintf(`%v("%v", %v) as %v`, funcName, f.Field, param, alias), nil
}
```